app/src/main/java/edu/uwstout/p2pchat/FileTransfer/SendDataService.java

```diff
@@ -154,23 +159,15 @@ private void transmitInMemoryFile(@NonNull final Intent intent)
 
             Log.d(LOG_TAG, "Client socket connected: " + socket.isConnected());
             OutputStream outputStream = socket.getOutputStream();
-            InputStream inputStream = null;
+            InputStream inputStream = serializeInMemoryFile(IMF);
 
-            try
+            // this is where the data is sent
+            if (inputStream != null)
```

Comment:
Maybe throw if inputStream is null?

Reply:
Unless we choose to have different behavior, having the function throw an exception just changes where the logging occurs. I could do it though.

Reply:
I just realized, it won't ever be null because it would only be null if serializeInMemoryFile threw an error, which means that we'd move to the catch anyways. Gonna remove the if altogether.